File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.py
# Last updated: 07/08/2025, 19:31:29
class Solution(object):
    def findGCD(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        small=min(nums)
        large=max(nums)
        for i in range(small,1,-1):
            if large%i==0 and small%i==0 :
                return i
        return 1

File: 2106-FindGreatestCommonDivisorOfArray/2106-FindGreatestCommonDivisorOfArray.cpp
// Last updated: 07/08/2025, 19:31:23
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int small=INT_MAX,large=INT_MIN;
        for(int i:nums)
            if(small>i) small=i;
        for(int i:nums)
            if(large<i) large=i;
        for(int i=small; i>0 ; i--)
            if(large%i==0 && small%i==0) return i;
        return 1;
    }
};